File: Ahorn/entities/InputFlagController.jl
module CommunalHelperInputFlagController

using ..Ahorn, Maple

@mapdef Entity "CommunalHelper/InputFlagController" Controller(
    x::Int,
    y::Int,
    flags::String="",
    toggle::Bool=true,
    resetFlags::Bool=false,
    delay::Float64=0.0,
    grabOverride::Bool=false,
)

const placements = Ahorn.PlacementDict(
    "Input Flag Controller (Communal Helper)" => Ahorn.EntityPlacement(
        Controller,
    ),
)

const sprite = "objects/CommunalHelper/inputFlagController/icon"

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::Controller) = Ahorn.drawSprite(ctx, sprite, 0, 0)

Ahorn.selection(entity::Controller) = Ahorn.getSpriteRectangle(sprite, Ahorn.position(entity)...)

end

File: Loenn/entities/misc/input_flag_controller.lua
local inputFlagController = {}

inputFlagController.name = "CommunalHelper/InputFlagController"
inputFlagController.depth = -1000000
inputFlagController.texture = "objects/CommunalHelper/inputFlagController/icon"

inputFlagController.placements = {
    {
        name = "controller",
        data = {
            flags = "",
            toggle = true,
            resetFlags = false,
            delay = 0.0,
            grabOverride = false
        }
    }
}

return inputFlagController
